src/Fixer/Casing/LowercaseStaticReferenceFixer.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Casing;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ */
+final class LowercaseStaticReferenceFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Class static references `self`, `static` and `parent` MUST be in lower case.',
+            [
+                new CodeSample('<?php
+class Foo extends Bar
+{
+    public function baz1()
+    {
+        return STATIC::baz2();
+    }
+
+    public function baz2($x)
+    {
+        return $x instanceof Self;
+    }
+
+    public function baz3(PaRent $x)
+    {
+        return true;
+    }
+}
+'),
+                new VersionSpecificCodeSample(
+                    '<?php
+class Foo extends Bar
+{
+    public function baz(?self $x) : SELF
+    {
+        return false;
+    }
+}
+',
+                    new VersionSpecification(71000)
+                ),
+            ]
+        );
+    }
+
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound([T_STATIC, T_STRING]);
+    }
+
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        for ($index = $tokens->count() - 1; $index >= 0; --$index) {
```

Comment:
I think this could be a `foreach ($tokens as $index => $token)`, yet I don't consider it blocking ;)

Reply:
Yes, it could and it is now :)

Reply:
thanks :D